src/generic/richmsgdlgg.cpp

```diff
@@ -77,6 +80,28 @@ void wxGenericRichMessageDialog::AddMessageDialogDetails(wxSizer *sizer)
         sizerDetails->Add( m_detailsPane, wxSizerFlags().Right().Expand() );
         sizer->Add( sizerDetails, 0, wxTOP|wxLEFT|wxRIGHT | wxALIGN_LEFT, 10 );
     }
+
+    if ( !m_footerText.empty() )
+    {
+        // add footer
+        sizer->Add( new wxStaticLine(this), wxSizerFlags().Expand().Border() );
+        wxSizer *footerSizer = new wxBoxSizer(wxHORIZONTAL);
+        if (m_footerIcon)
+        {
+            wxSize iconSize = wxArtProvider::GetNativeSizeHint(wxART_MESSAGE_BOX);
+            if ( iconSize.IsFullySpecified() )
+                iconSize /= 2;
```

Comment:
Why do we do this? Shouldn't we use a different argument to `GetNativeSizeHint()` instead, e.g. `wxART_MENU` perhaps?

I think it at least merits a comment.